From d470e77faa78018b8ec8ebb9de91dab153b8b1d0 Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
Date: Mon, 27 Aug 2018 01:33:25 +0900
Subject: [PATCH 2/8] tools/sppc: dockerfile for custom SPP repo

This update is to change dockerfiles of SPP for applying the change of
build script for using any of git repository.

Signed-off-by: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
---
 tools/sppc/build/ubuntu/spp/Dockerfile.16.04  | 12 +++++++-----
 tools/sppc/build/ubuntu/spp/Dockerfile.18.04  | 12 +++++++-----
 tools/sppc/build/ubuntu/spp/Dockerfile.latest | 11 ++++++-----
 3 files changed, 20 insertions(+), 15 deletions(-)

diff --git a/tools/sppc/build/ubuntu/spp/Dockerfile.16.04 b/tools/sppc/build/ubuntu/spp/Dockerfile.16.04
index cceb7e3..b55f056 100644
--- a/tools/sppc/build/ubuntu/spp/Dockerfile.16.04
+++ b/tools/sppc/build/ubuntu/spp/Dockerfile.16.04
@@ -7,6 +7,7 @@ ARG dpdk_repo
 ARG dpdk_branch
 ARG spp_repo
 ARG spp_branch
+ARG spp_dir
 
 ENV PATH ${rte_sdk}/${rte_target}/app:${PATH}
 ENV http_proxy ${http_proxy}
@@ -14,6 +15,7 @@ ENV https_proxy $https_proxy
 ENV no_proxy ${no_proxy}
 ENV RTE_SDK ${rte_sdk}
 ENV RTE_TARGET ${rte_target}
+ENV SPP_DIR ${spp_dir}
 
 RUN apt-get update && apt-get install -y \
     git \
@@ -31,14 +33,14 @@ RUN apt-get update && apt-get install -y \
     && rm -rf /var/lib/apt/lists/*
 
 WORKDIR $home_dir
-RUN git clone $dpdk_branch $dpdk_repo
-RUN git clone $spp_branch $spp_repo
+RUN git clone ${dpdk_branch} ${dpdk_repo}
+RUN git clone ${spp_branch} ${spp_repo}
 
 # Compile DPDK and SPP
-WORKDIR $rte_sdk
-RUN make install T=$rte_target
+WORKDIR ${rte_sdk}
+RUN make install T=${rte_target}
 
-WORKDIR ${home_dir}/spp
+WORKDIR ${home_dir}/${spp_dir}
 RUN make
 
 # Set working directory when container is launched
diff --git a/tools/sppc/build/ubuntu/spp/Dockerfile.18.04 b/tools/sppc/build/ubuntu/spp/Dockerfile.18.04
index 51a0902..8377623 100644
--- a/tools/sppc/build/ubuntu/spp/Dockerfile.18.04
+++ b/tools/sppc/build/ubuntu/spp/Dockerfile.18.04
@@ -7,6 +7,7 @@ ARG dpdk_repo
 ARG dpdk_branch
 ARG spp_repo
 ARG spp_branch
+ARG spp_dir
 
 ENV PATH ${rte_sdk}/${rte_target}/app:${PATH}
 ENV http_proxy ${http_proxy}
@@ -14,6 +15,7 @@ ENV https_proxy $https_proxy
 ENV no_proxy ${no_proxy}
 ENV RTE_SDK ${rte_sdk}
 ENV RTE_TARGET ${rte_target}
+ENV SPP_DIR ${spp_dir}
 
 RUN apt-get update && apt-get install -y \
     git \
@@ -31,14 +33,14 @@ RUN apt-get update && apt-get install -y \
     && rm -rf /var/lib/apt/lists/*
 
 WORKDIR $home_dir
-RUN git clone $dpdk_branch $dpdk_repo
-RUN git clone $spp_branch $spp_repo
+RUN git clone ${dpdk_branch} ${dpdk_repo}
+RUN git clone ${spp_branch} ${spp_repo}
 
 # Compile DPDK and SPP
-WORKDIR $rte_sdk
-RUN make install T=$rte_target
+WORKDIR ${rte_sdk}
+RUN make install T=${rte_target}
 
-WORKDIR ${home_dir}/spp
+WORKDIR ${home_dir}/${spp_dir}
 RUN make
 
 # Set working directory when container is launched
diff --git a/tools/sppc/build/ubuntu/spp/Dockerfile.latest b/tools/sppc/build/ubuntu/spp/Dockerfile.latest
index 3142c11..3b96080 100644
--- a/tools/sppc/build/ubuntu/spp/Dockerfile.latest
+++ b/tools/sppc/build/ubuntu/spp/Dockerfile.latest
@@ -7,6 +7,7 @@ ARG dpdk_repo
 ARG dpdk_branch
 ARG spp_repo
 ARG spp_branch
+ARG spp_dir
 
 ENV PATH ${rte_sdk}/${rte_target}/app:${PATH}
 ENV http_proxy ${http_proxy}
@@ -31,14 +32,14 @@ RUN apt-get update && apt-get install -y \
     && rm -rf /var/lib/apt/lists/*
 
 WORKDIR $home_dir
-RUN git clone $dpdk_branch $dpdk_repo
-RUN git clone $spp_branch $spp_repo
+RUN git clone ${dpdk_branch} ${dpdk_repo}
+RUN git clone ${spp_branch} ${spp_repo}
 
 # Compile DPDK and SPP
-WORKDIR $rte_sdk
-RUN make install T=$rte_target
+WORKDIR ${rte_sdk}
+RUN make install T=${rte_target}
 
-WORKDIR ${home_dir}/spp
+WORKDIR ${home_dir}/${spp_dir}
 RUN make
 
 # Set working directory when container is launched
-- 
2.7.4

